### PR TITLE
Fix lazy decomposition

### LINKED ISSUE #1858: rPCA inconsistency?
Hi all,

recently I may have spotted an inconsistency in Hyperspy's implementation of robust PCA. Subsequently I tried to narrow it down in an idealized test case and keep getting the same strange behavior. So the test case consists of a matrix ```A``` full of ones, except for one element, which is set to two. In my understanding, this should be decomposed into a matrix ```L``` filled with ones and a matrix ```S``` filled with zeros and a one in the element corresponding to the two in ```A```, such that approximately ```L+S = A```. I checked this with another implementation (```r_pca``` from https://github.com/dganguli/robust-pca), which gave the expected result, but Hyperspy disagreed completely. Is this a bug or am I using Hyperspy's rPCA routine incorrectly? I'm using Hyperspy version 1.3. Here's my code:

```python
import numpy as np

import importlib
importlib.import_module('r_pca')
from r_pca import R_pca

import hyperspy.api as hs

n=5
m=10
A = np.ones((n,m))
A[int(n/2),int(m/2)] = 2.0

rpca = R_pca(A)
L, S = rpca.fit(max_iter=1000, iter_print=100)

hsdata = hs.signals.Signal1D(A)
Lhs,Shs = hsdata.decomposition(algorithm='RPCA_GoDec', return_info=True, output_dimension=min(np.shape(A)), power=3)

print('Ganguli-Candes: L= ')
print(L)

print('Hyperspy: L= ')
print(Lhs)

print('Ganguli-Candes: S= ')
print(S)

print('Hyperspy: S= ')
print(Shs)

print('Ganguli-Candes: L+S= ')
print(L+S)

print('Hyperspy: L+S= ')
print(Lhs+Shs)

print('Done.')

import hyperspy
print('Hyperspy version '+str(hyperspy.__version__))
```

And here's the corresponding output:
```
iteration: 1, error: 18.36113064260148
iteration: 8, error: 1.3991493881412786e-06
WARNING:hyperspy.learn.rpca:Threshold 'lambda1' is set to default: 1 / sqrt(nsamples)
WARNING:hyperspy.learn.rpca:Convergence tolerance not specifed. Defaulting to 1e-3
WARNING:hyperspy.learn.rpca:Maximum iterations not specified. Defaulting to 1e3
Ganguli-Candes: L=
[[1.00010184 1.00010184 1.00010184 1.00010184 1.00010184 1.00019043
  1.00010184 1.00010184 1.00010184 1.00010184]
 [1.00010184 1.00010184 1.00010184 1.00010184 1.00010184 1.00019043
  1.00010184 1.00010184 1.00010184 1.00010184]
 [1.00031282 1.00031282 1.00031282 1.00031282 1.00031282 1.00040142
  1.00031282 1.00031282 1.00031282 1.00031282]
 [1.00010184 1.00010184 1.00010184 1.00010184 1.00010184 1.00019043
  1.00010184 1.00010184 1.00010184 1.00010184]
 [1.00010184 1.00010184 1.00010184 1.00010184 1.00010184 1.00019043
  1.00010184 1.00010184 1.00010184 1.00010184]]
Hyperspy: L=
[[1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 2. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]]
Ganguli-Candes: S=
[[0.         0.         0.         0.         0.         0.
  0.         0.         0.         0.        ]
 [0.         0.         0.         0.         0.         0.
  0.         0.         0.         0.        ]
 [0.         0.         0.         0.         0.         0.99959858
  0.         0.         0.         0.        ]
 [0.         0.         0.         0.         0.         0.
  0.         0.         0.         0.        ]
 [0.         0.         0.         0.         0.         0.
  0.         0.         0.         0.        ]]
Hyperspy: S=
[[1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
 [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]]
Ganguli-Candes: L+S=
[[1.00010184 1.00010184 1.00010184 1.00010184 1.00010184 1.00019043
  1.00010184 1.00010184 1.00010184 1.00010184]
 [1.00010184 1.00010184 1.00010184 1.00010184 1.00010184 1.00019043
  1.00010184 1.00010184 1.00010184 1.00010184]
 [1.00031282 1.00031282 1.00031282 1.00031282 1.00031282 2.
  1.00031282 1.00031282 1.00031282 1.00031282]
 [1.00010184 1.00010184 1.00010184 1.00010184 1.00010184 1.00019043
  1.00010184 1.00010184 1.00010184 1.00010184]
 [1.00010184 1.00010184 1.00010184 1.00010184 1.00010184 1.00019043
  1.00010184 1.00010184 1.00010184 1.00010184]]
Hyperspy: L+S=
[[2. 2. 2. 2. 2. 2. 2. 2. 2. 2.]
 [2. 2. 2. 2. 2. 2. 2. 2. 2. 2.]
 [2. 2. 2. 2. 2. 3. 2. 2. 2. 2.]
 [2. 2. 2. 2. 2. 2. 2. 2. 2. 2.]
 [2. 2. 2. 2. 2. 2. 2. 2. 2. 2.]]
Done.
Hyperspy version 1.3
```

Just out of curiosity I also tried the same with larger dimensions and with the transpose of ```A```, but the result was equivalent, as it should be.

### PR DESCRIPTION
**Pull request by francisco-dlp**
### Description of the change

Resubmission of #1878 and #1802 because it was not possible to test #1802 withour #1878.

* Enables ``unfold`` for lazy signals
* adds "svd" to the lazy decomposition algorithms and make it default (as in its in-memory counterpart)
* Adds tests for ``get_*_model``
* Add a warning to the Big data chapter explaining that the features are in beta state.
* Fix #1858
* Add ``compute`` keyword to ``crop_decomposition_result``.

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] update docstring (if appropriate),
- [x] update user guide (if appropriate),
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

File: hyperspy/_signals/lazy.py [hunk 1]
```diff
@@ -772,16 +777,20 @@ def decomposition(self,
 
             # REPROJECT
             if reproject:
+                print(reproject)
```
Comment (ericpre) [inline hyperspy/_signals/lazy.py]: Some leftover?

**Review by ericpre — COMMENTED** [review]
One small `print` to remove and it's good to go!

File: hyperspy/_signals/lazy.py (hunk 1 above)
Comment (francisco-dlp) [reply hyperspy/_signals/lazy.py]: Indeed, thanks!

**Comment by ericpre** [comment]
There is still a failure which I don't think is related to this PR:
https://travis-ci.org/hyperspy/hyperspy/jobs/368618591

I think that I have seen it before and it seems to happen fairly rarely (~<10 %).

**Comment by ericpre** [comment]
Sorry I restarted the build... it would have good to keep it for the record...

Below is the failing tests 
```python
def test_corrupted_robust():
    W, H = onmf(Y, r, robust=True)
    res = compare(np.dot(W, H), X.T)
    print(res)
    assert res < 0.095
```
with a `res` value of ~0.0957

**Comment by francisco-dlp** [comment]
If it's known to fail randomly I'll update the figure, thanks!

**Comment by francisco-dlp** [comment]
Done!

**Pull request merged**